flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierUnaligner.java

```diff
@@ -404,8 +400,14 @@ synchronized int getNumOpenChannels() {
 			return numOpenChannels;
 		}
 
+		@VisibleForTesting
 		synchronized long getCurrentCheckpointId() {
 			return currentReceivedCheckpointId;
 		}
+
+		@VisibleForTesting
+		boolean isCheckpointPending() {
```

Comment:
A bit unrelated to your change, but shouldn't that also be synchronized?
Also is it necessary to move the method?

Reply:
The only purpose of retaining this method now is for unit tests. For the main codes, all the callers are already in `synchronized`, and for test usages it has no race condition issues. So I am a bit torn whether to add `synchronized` for it, WDYT?

Reply:
okay, fair enough. let's keep it this way.

Reply:
Actually I already refactored the main codes to not use this method which only touched by tests now.

Reply:
I'd prefer to use this method in main code instead of spreading the knowledge in many places. If I read `isCheckpointPending`, I understand it much quicker than numBarriersReceived > 0.
But it's more a personal taste, so up to you.